mtbl/writer.c

```diff
@@ -155,14 +198,16 @@ mtbl_writer_init(const char *fname, const struct mtbl_writer_options *opt)
 void
 mtbl_writer_destroy(struct mtbl_writer **w)
 {
-	if (*w != NULL) {
+	if (*w) {
 		if (!(*w)->closed) {
 			_mtbl_writer_finish(*w);
 			close((*w)->fd);
 		}
+
 		block_builder_destroy(&((*w)->data));
 		block_builder_destroy(&((*w)->index));
 		ubuf_destroy(&(*w)->last_key);
+
 		free(*w);
```

Comment:
all of the free(x); x = null; sequences could be
replaced by my_free as in:
libmy/my_alloc.h:#define my_free(ptr) do { free(ptr); (ptr) = NULL; } while (0)

Reply:
Chris requested that I minimize diffs unless functionality changes significantly, however because you specifically requested it I would be happy to change this.

Reply:
follow Chris, please